### PR TITLE
Get All Flights for User Endpoint 

### LINKED ISSUE #1: Get User Flights Endpoint
create endpoint to gather all of an individual user's flights

### PR DESCRIPTION
**Pull request by cariperi**
## Neccesary checkmarks:

    [x] All Tests are Passing in all environments

    [x] The code will run locally

## Type of change

    [x] New feature
    [] Bug Fix
    [] Refactor

## Description of Change:

    description:  This PR closes #1  and adds an endpoint for Get All Flights for User with happy and sad paths, and adds tests for returning flights for a user, returning an empty array for a user with no flights (edge case) and sad path where user is not found.  

Note that these files include commented-out code with a different approach to serialization, using JSON-API and the class-based views.  This is included for our reference / learning, but the solution committed here is aligned with the team's approach for manual serialization. 
    
## Requested feedback: 
    
    I'd like feedback on... testing strategies - I followed @KatieLonsdale's approach for testing specific values and included tests for one possible edge case, but curious if there is anything else you think I should include.

## Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

## Testing Changes

    [] No Tests have been changed
    [x] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

    [] My code has no unused/commented out code. --- see note above
    [x] My code has no breakpoints
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link: #1

### CODE REVIEW

File: flight_log_be/serializers.py [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
 from .models import User
+from .models import Flight
+
 
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ['id', 'first_name', 'last_name']
\ No newline at end of file
+        fields = ["id", "first_name", "last_name"]
```
Comment (KatieLonsdale) [inline flight_log_be/serializers.py]: I took id out of my fields in the user serializer because I didn't want it to show up in the attributes. Did you get it to not show up?

File: flight_log_be/settings.py [hunk 2]
```diff
@@ -122,9 +135,44 @@
...
-STATIC_URL = 'static/'
+STATIC_URL = "static/"
 
 # Default primary key field type
 # https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field
 
-DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
+
+# # REST Framework Settings for REST Framework JSON API
```
Comment (KatieLonsdale) [inline flight_log_be/settings.py:145]: Really hoping we have time to figure this out once we get the endpoints done!

**Review by KatieLonsdale — APPROVED** [review]
Nice job!

File: flight_log_be/serializers.py (hunk 1 above)
Comment (cariperi) [reply flight_log_be/serializers.py]: Nope, this was old - we removed it in the merge conflict!  I did figure out how to get it to now show up in Flights (since Brandon needs it to show up in his response) but it's very manual!

File: flight_log_be/tests/test_views.py [hunk 3]
```diff
@@ -2,6 +2,136 @@
...
+
+        self.assertEqual(flight_response_1["type"], "flight")
+        self.assertEqual(flight_response_2["type"], "flight")
+
+        attributes = flight_response_1["attributes"]
+        self.assertEqual(attributes["date"], flight_1.date.isoformat())
+        self.assertEqual(attributes["start_location"], flight_1.start_location)
+        self.assertEqual(attributes["end_location"], flight_1.end_location)
+        self.assertEqual(attributes["day_hours"], f"{flight_1.day_hours}")
+        self.assertEqual(attributes["night_hours"], f"{flight_1.night_hours}")
```
Comment (brenicillin) [inline flight_log_be/tests/test_views.py:104]: This was a doozy, great job figuring it out!

File: flight_log_be/urls.py [hunk 4]
```diff
@@ -17,9 +17,12 @@
 from django.contrib import admin
 from django.urls import path
 from flight_log_be import views
+from rest_framework.urlpatterns import format_suffix_patterns
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('api/v1/users/<int:id>', views.return_a_user),
-    path('api/v1/users/<int:user>/flights/', views.create_flight, name='create_flight')
+    path('api/v1/users/<int:user>/flights/', views.flights),
```
Comment (brenicillin) [inline flight_log_be/urls.py:25]: great job consolidating our paths!

**Review by brenicillin — APPROVED** [review]
